### PR TITLE
Add rustfmt that matches comit-rs

### PR DESCRIPTION
**Pull request by rishflab**
Added rustfmt config that matches comit-rs. Formatted the project. I am hoping the ci.yml changes I made allow me to check the formatting using the nightly toolchain in the CI.

We can freeze this repo after this is merged and move it into comit-rs @D4nte @thomaseizinger

### CODE REVIEW

**Comment by D4nte** [comment]
Please note I doing some of the changes present in this PR already with https://github.com/coblox/nectar/pull/116

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -39,7 +38,7 @@ jobs:
         run: cargo tomlfmt -d -p Cargo.toml
 
       - name: Check code formatting
-        run: cargo fmt -- --check
+        run: cargo +nightly fmt -- --check
```
Comment (D4nte) [inline .github/workflows/ci.yml:41]: You need to install the nightly toolchain (and the stable).

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -20,8 +20,8 @@ jobs:
 
       - name: Install Rust
         uses: actions-rs/toolchain@v1
-        id: toolchain
         with:
+          toolchain: nightly
```
Comment (D4nte) [inline .github/workflows/ci.yml:24]: You need to install *both* nightly and stable.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -36,10 +43,10 @@ jobs:
         run: cargo tomlfmt -d -p Cargo.toml
 
       - name: Check code formatting
-        run: cargo fmt -- --check
+        run: cargo +nightly-2020-01-15-x86_64-unknown-linux-gnu fmt -- --check
 
       - name: Run clippy
-        run: cargo clippy --all-targets -- -D warnings
+        run: cargo +nightly-2020-01-15-x86_64-unknown-linux-gnu clippy --all-targets -- -D warnings
```
Comment (thomaseizinger) [inline .github/workflows/ci.yml:49]: We run clippy on stable Rust in comit-rs :)

That is why your build is failing!

**Review by thomaseizinger — APPROVED** [review]

**Pull request merged**